pkg/virt-handler/cache/domain-watcher_test.go

```diff
@@ -0,0 +1,51 @@
+package cache
+
+import (
+	"os"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	cmdclient "kubevirt.io/kubevirt/pkg/virt-handler/cmd-client"
+)
+
+var _ = Describe("Domain Watcher", func() {
+	Context("listSockets ", func() {
+		It("Ignore sockets that aren't in ghost record cache", func() {
+			podsDir, err := os.MkdirTemp("", "")
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(os.RemoveAll, podsDir)
+
+			ghostCacheDir, err := os.MkdirTemp("", "")
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(os.RemoveAll, ghostCacheDir)
+
+			ghostRecordStore := InitializeGhostRecordCache(NewIterableCheckpointManager(ghostCacheDir))
+
+			cmdclient.SetPodsBaseDir(podsDir)
+
+			const nonExistingDomainPodUID = "1234"
+			nonExistingDomainSocketPath := cmdclient.SocketFilePathOnHost(nonExistingDomainPodUID)
+			Expect(os.MkdirAll(filepath.Dir(nonExistingDomainSocketPath), 0755)).To(Succeed())
+			f1, err := os.Create(nonExistingDomainSocketPath)
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(f1.Close)
+
+			const existingDomainPodUID = "5678"
+			existingDomainSocketPath := cmdclient.SocketFilePathOnHost(existingDomainPodUID)
+			Expect(os.MkdirAll(filepath.Dir(existingDomainSocketPath), 0755)).To(Succeed())
+			f2, err := os.Create(existingDomainSocketPath)
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(f2.Close)
+			err = ghostRecordStore.Add("test-ns", "test-domain", existingDomainSocketPath, existingDomainPodUID)
+			Expect(err).ToNot(HaveOccurred())
+
+			socketFiles, err := listSockets(GhostRecordGlobalStore.list())
```

Comment:
@enp0s3 why are you calling GhostRecordGlobalStore.list() here instead of ghostRecordStore.list()? Isn't GhostRecordGlobalStore.list()  just empty?